File: Hard/2026-06-05-3753-Total-Waviness-of-Numbers-in-Range-II/solution.java
public class Solution {
    public long totalWaviness(long num1, long num2) {
        long totalWaviness = 0;
        for (long i = num1; i <= num2; i++) {
            totalWaviness += calculateWaviness(i);
        }
        return totalWaviness;
    }

    private int calculateWaviness(long number) {
        String numStr = Long.toString(number);
        int waviness = 0;
        int length = numStr.length();
        
        if (length < 3) {
            return 0;
        }
        
        for (int j = 1; j < length - 1; j++) {
            char left = numStr.charAt(j - 1);
            char mid = numStr.charAt(j);
            char right = numStr.charAt(j + 1);
            
            if (mid > left && mid > right) {
                waviness++; // peak
            } else if (mid < left && mid < right) {
                waviness++; // valley
            }
        }
        
        return waviness;
    }
}

File: Hard/2026-06-05-3753-Total-Waviness-of-Numbers-in-Range-II/solution.js
var totalWaviness = function(num1, num2) {
    let totalWaviness = 0;

    for (let i = num1; i <= num2; i++) {
        const strNum = i.toString();
        const len = strNum.length;

        if (len < 3) continue;

        let waviness = 0;

        for (let j = 1; j < len - 1; j++) {
            if (strNum[j] > strNum[j - 1] && strNum[j] > strNum[j + 1]) {
                waviness++;
            } else if (strNum[j] < strNum[j - 1] && strNum[j] < strNum[j + 1]) {
                waviness++;
            }
        }

        totalWaviness += waviness;
    }

    return totalWaviness;
};